### PR TITLE
Created some initial tests for the add imports evaluator

### PR DESCRIPTION
**Pull request by wahleric**

### CODE REVIEW

File: backend_plugin/src/test/java/negatives/evaluators/AddImportsNegative.java [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+	 * Tests the user copy-pasting an import statement into the document
+	 * when there are unresolved variables
+	 */
+	@Test
+	public void unresolvedTypesAddImportStatementWithCopyPaste() {
+		try {
+			am.addAnnotation(ANNOTATION, POSITION);
+			eval.evaluateAnnotationModelChanges(am);
+			int offset = 0;
+			assertTrue(mockUserInput(MOCK_IMPORT, offset));
```
Comment (johnbarUW) [inline backend_plugin/src/test/java/negatives/evaluators/AddImportsNegative.java:122]: We do catch this case? Awesome.

File: backend_plugin/src/test/java/negatives/evaluators/AddImportsNegative.java [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+		}
+	}
+	
+	/**
+	 * Tests the user typing "import " inside a comment line, rather than as
+	 * an import statement at the start of a line, when unresolved variables
+	 * exist
+	 */
+	@Test
+	public void unresolvedVariablesImportInComment() {
```
Comment (johnbarUW) [inline backend_plugin/src/test/java/negatives/evaluators/AddImportsNegative.java:161]: Good test case.

**Review by johnbarUW — APPROVED** [review]
Reasonable test cases. We can copy the false negative tests into a separate test to hep with our success metrics

File: backend_plugin/src/test/java/negatives/evaluators/AddImportsNegative.java [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+	/**
+	 * Tests the user starting a valid import statement by typing
+	 * "import " one character at a time when there are no unresolved
+	 * variables
+	 */
+	@Test
+	public void noUnresolvedTypesAddImportOneCharacterAtATime() {
+		try {
+			int offset = 0;
+			assertFalse(mockUserInput("i", offset++));
```
Comment (DavidThien) [inline backend_plugin/src/test/java/negatives/evaluators/AddImportsNegative.java]: To make this more readable, you can use `for (char c : "import".toCharArray())`

**Review by DavidThien — DISMISSED** [review]
Looks good overall. One thing to think about for the future is how we can simplify having to write all of these test cases down. Particularly for stuff like this where we are really just simulating some typing, it would be great to have an easier way to mock this up.

File: backend_plugin/src/test/java/negatives/evaluators/AddImportsNegative.java (hunk 3 above)
Comment (wahleric) [reply backend_plugin/src/test/java/negatives/evaluators/AddImportsNegative.java]: Good call, I've refactored the class to simplify these actions as noted

**Comment by wahleric** [comment]
I refactored the class per David's comment, and also created a second test file under the *Test.java convention for tests that should be run on CI

**Pull request merged**